docs/WS_ORDER_EVENTS.md

```diff
@@ -0,0 +1,31 @@
+# Websocket trade events
+
+GoCryptoTrader unifies trades and order update events by composing an
+order.Detail object.  This is the full list of order.Detail fields
+that exchange implementations should populate on streamed
+trade/order-update events.  As exchanges provide different APIs, not
+all fields are mandatory.
+
+```
+| order.Detail field   | Description                                                       | Condition                                               | Presence  |
+|----------------------+-------------------------------------------------------------------+---------------------------------------------------------+-----------|
+| Price                | Original price assigned to order                                  | Depends on order type (e.g. limit orders have prices)   | Mandatory |
+| Amount               | Original quantity assigned to order                               |                                                         | Mandatory |
+| AverageExecutedPrice | Average price of what's traded thus far                           | Order is filled, partially filled or partially canceled | Mandatory |
```

Comment:
I think this one should be optional. It is not included in all exchange API responses and potentially can be calculated on our side if we gather the trades

Reply:
Sounds painful to exclude it. :(